Modules/Journey/tabs/QuestsByZone/QuestsByZone.lua

```diff
@@ -38,6 +37,58 @@ function _QuestieJourney.questsByZone:ManageTree(container, zoneTree)
     zoneTreeFrame:SetTree(zoneTree)
 
     zoneTreeFrame.treeframe:SetWidth(415)
+
+    if _QuestieJourney.lastZoneSelection[3] then
+        local savedSelection = _QuestieJourney.lastZoneSelection[3]
+        local sel, questId = strsplit("\001", savedSelection)
+
+        if questId then
+            questId = tonumber(questId)
+            local questExists = false
+
+            for _, category in pairs(zoneTree) do
+                if category.children then
+                    for _, quest in pairs(category.children) do
+                        if quest.value and quest.value == questId then
+                            questExists = true
+                            break
+                        end
+                    end
+                end
+                if questExists then break end
+            end
+
+            if questExists then
+                zoneTreeFrame:SetSelected(sel, savedSelection)
+
+                C_Timer.After(0.1, function()
+                    local questIdNum = tonumber(questId)
+                    if questIdNum then
+                        local quest = QuestieDB.GetQuest(questIdNum)
+                        if quest then
+
+                            local master = zoneTreeFrame.frame.obj
+                            master:ReleaseChildren()
+                            master:SetLayout("fill")
+                            master:SetFullWidth(true)
+                            master:SetFullHeight(true)
+
+                            ---@class ScrollFrame
+                            local scrollFrame = AceGUI:Create("ScrollFrame")
+                            scrollFrame:SetLayout("flow")
+                            scrollFrame:SetFullHeight(true)
+                            master:AddChild(scrollFrame)
+
+                            _QuestieJourney:DrawQuestDetailsFrame(scrollFrame, quest)
+                        end
+                    end
+                end)
+            else
+                _QuestieJourney.lastZoneSelection[3] = nil
+            end
+        end
```

Comment:
Maybe this can be a function, so we have a proper name for the code? Just to make it clear what this code is for (which is re-selecting the previous selection, right?)

Also this is very similar now to the code in `_QuestieJourney:RestoreQuestSelection` 🤔